### DISCUSSION TITLE
Support for the game controller standard: gamepads, joysticks, flightsim throttle and pedal gear etc.

### DISCUSSION THREAD

**Discussion by drjedd** [helgoboss/helgobox#1191] [Ideas]
Hello,

As you requested I'm opening a discussion here.

The idea is to extend Realearn's functionality to detect plugged in (or bluetooth) controllers and use the axises and buttons on these (which have a common interface in Windows at least) to send midi notes or CC data and whatever else is relevant.
This would allow the mapping of custom devices like a joystick to control cutoff/resonance or throttle levers mapped to expression or dynamics, etc. it would also greatly extend the accessibility of Reaper since there are thousands of shapes and functions in these devices and the whole Xbox Adaptive Controller platform uses this interface: https://www.xbox.com/en-US/accessories/controllers/xbox-adaptive-controller

I'm writing this having no idea if it's worth it: from within games those controllers are quite precise but I don't know how the translation to midi would work. There is a utility out there called Fergo JoystickMidi : https://www.fergonez.net/softwares/fjoymidi
But it has drawbacks: it stops working if the tool's window is not focused anymore, it doesn't allow for any customization like inverting an axis etc. but I'm linking it here are as a point of reference I guess.
(I tried to get in touch with the author but got no response so far. The project itself has not seen updates for 14 years.)

Thanks, have a lovely day!

jedd

**Comment by helgoboss** [comment]
Just a hint: REAPER itself already supports joysticks. I guess you know that already and would like to see joysticks explicitly in ReaLearn?

**Reply by drjedd** [reply]
If I knew about this then I forgot a long time ago! Thanks for the heads up.

This does require some scripting every time you want to map it differently though? Which Realearn would make a little bit faster and easier it seems?

**Reply by drjedd** [reply]
To give you an actual example, I wanted to experiment with something like this over regular faders to control the usual strings CCs : Dynamics/Expression/Vibrato

https://www.youtube.com/watch?v=ocP2AiizxCE

(Although I have no idea how well it would work, both ergonomically and on the sensor/software side, but that's why we're experimenting 🙃 )

**Reply by helgoboss** [reply]
Yes, just wanted to mention it. With ReaLearn you get a lot more than the native possibilities, just as with MIDI control. Might even add feedback (vibration ;)? Let's see. It's not really something I plan to add quickly, but it sounds like fun. Will look into it a bit when I'm in the mood.

**Reply by drjedd** [reply]
Awesome! Yeah there is no rush I understand and it's great (and inspiring) to see you working on the project with this mindset!
Vibration support sounds cool but a barebones mapping of buttons to notes and axises to CCs would already be amazing.
